src/repl/scala/tools/nsc/interpreter/MemberHandlers.scala

```diff
@@ -214,27 +214,34 @@ trait MemberHandlers {
     val Import(expr, selectors) = imp
     def targetType = intp.global.rootMirror.getModuleIfDefined("" + expr) match {
       case NoSymbol => intp.typeOfExpression("" + expr)
-      case sym      => sym.thisType
+      case sym      => sym.moduleClass.thisType
```

Comment:
Good catch! 
```
    /** If symbol is a class, the type `this.type` in this class,
     * otherwise `NoPrefix`.
     * We always have: thisType <:< typeOfThis
     */
    def thisType: Type = NoPrefix
```

From the repl's `:power` mode:
```
scala> object o
defined object o

scala> val module = typeOf[o.type].termSymbol
module: $r.intp.global.Symbol = object o

scala> module.thisType
res7: $r.intp.global.Type = <noprefix>

scala> module.moduleClass.thisType
res8: $r.intp.global.Type = o.type
```

However, this can be had more directly:
```
scala> module.tpe
res12: $r.intp.global.Type = o.type
```